### ISSUE TITLE
exception in printing types within semanticdb-scalac

### ISSUE THREAD

**Issue by MasseGuillaume** [scalameta/scalameta#1116]
```scala
package com.olegych.scastie.api

import play.api.libs.json._

sealed trait EnsimeServerState
object EnsimeServerState {
  case object Unknown extends EnsimeServerState {
    override def toString: String = "Unknown"
  }

  case object Initializing extends EnsimeServerState {
    override def toString: String = "Initializing"
  }

  case object CreatingConfig extends EnsimeServerState {
    override def toString: String = "CreatingConfig"
  }

  case object Connecting extends EnsimeServerState {
    override def toString: String = "Connecting"
  }

  case object Indexing extends EnsimeServerState {
    override def toString: String = "Indexing"
  }

  case object Ready extends EnsimeServerState {
    override def toString: String = "Ready"
  }

  implicit object EnsimeServerStateFormat extends Format[EnsimeServerState] {
    def writes(state: EnsimeServerState): JsValue = {
      JsString(state.toString)
    }

    private val values: Map[String, EnsimeServerState] =
      List(
        Unknown,
        Initializing,
        CreatingConfig,
        Connecting,
        Indexing,
        Ready
      ).map(v => (v.toString, v)).toMap

    def reads(json: JsValue): JsResult[EnsimeServerState] = {
      json match {
        case JsString(tpe) => {
          values.get(tpe) match {
            case Some(v) => JsSuccess(v)
            case _       => JsError(Seq())
          }
        }
        case _ => JsError(Seq())
      }
    }
  }
}
```

I ran scalafmt on this file and I got the following stacktrace

NB: this part is causing the issue:

```scala
 List(
        Unknown,
        Initializing,
        CreatingConfig,
        Connecting,
        Indexing,
        Ready
      )
```

```
[warn] failed to generate semanticdb for /home/gui/scastie/api/src/main/scala/com.olegych.scastie.api/EnsimeServerState.scala:
[warn] org.scalameta.UnreachableError: this code path should've been unreachable
[warn] where tpe = Product with Serializable with com.olegych.scastie.api.EnsimeServerState
[warn]  at org.scalameta.UnreachableError$.raise(package.scala:40)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.printType(PrinterOps.scala:406)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$printType$5(PrinterOps.scala:399)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$printType$5$adapted(PrinterOps.scala:399)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$printType$1(PrinterOps.scala:344)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$printType$1$adapted(PrinterOps.scala:340)
[warn]  at scala.collection.immutable.List.foreach(List.scala:389)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.wrapped$1(PrinterOps.scala:340)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.printType(PrinterOps.scala:399)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.processTreePrinting(PrinterOps.scala:801)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.printTree(PrinterOps.scala:415)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.$anonfun$print$1(Printers.scala:33)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.$anonfun$print$1$adapted(Printers.scala:17)
[warn]  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
[warn]  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
[warn]  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.print(Printers.scala:17)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.super$print(PrinterOps.scala:55)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$print$1(PrinterOps.scala:60)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$print$1$adapted(PrinterOps.scala:52)
[warn]  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
[warn]  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
[warn]  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.print(PrinterOps.scala:52)
[warn]  at scala.reflect.internal.Printers$TreePrinter.$anonfun$printRow$1(Printers.scala:106)
[warn]  at scala.reflect.internal.Printers$TreePrinter.$anonfun$printRow$1$adapted(Printers.scala:106)
[warn]  at scala.reflect.internal.Printers$TreePrinter.printSeq(Printers.scala:96)
[warn]  at scala.reflect.internal.Printers$TreePrinter.printRow(Printers.scala:106)
[warn]  at scala.reflect.internal.Printers$TreePrinter.printTree(Printers.scala:436)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.processTreePrinting(PrinterOps.scala:681)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.printTree(PrinterOps.scala:415)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.$anonfun$print$1(Printers.scala:33)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.$anonfun$print$1$adapted(Printers.scala:17)
[warn]  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
[warn]  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
[warn]  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.print(Printers.scala:17)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.super$print(PrinterOps.scala:55)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$print$1(PrinterOps.scala:60)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$print$1$adapted(PrinterOps.scala:52)
[warn]  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
[warn]  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
[warn]  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.print(PrinterOps.scala:52)
[warn]  at scala.meta.internal.semanticdb.PrinterOps.showSynthetic(PrinterOps.scala:22)
[warn]  at scala.meta.internal.semanticdb.PrinterOps.showSynthetic$(PrinterOps.scala:19)
[warn]  at scala.meta.internal.SemanticdbPlugin.showSynthetic(ScalahostPlugin.scala:9)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.$anonfun$tryFindInferred$4(DocumentOps.scala:332)
[warn]  at scala.collection.immutable.List.map(List.scala:283)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.tryFindInferred(DocumentOps.scala:332)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:391)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:149)
[warn]  at scala.reflect.internal.Trees.$anonfun$itraverse$1(Trees.scala:1220)
[warn]  at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
[warn]  at scala.reflect.internal.Trees.traverseMemberDef$1(Trees.scala:1214)
[warn]  at scala.reflect.internal.Trees.itraverse(Trees.scala:1339)
[warn]  at scala.reflect.internal.Trees.itraverse$(Trees.scala:1211)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:398)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:149)
[warn]  at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$2(Trees.scala:2498)
[warn]  at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
[warn]  at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$1(Trees.scala:2498)
[warn]  at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
[warn]  at scala.reflect.internal.Trees.traverseComponents$1(Trees.scala:1243)
[warn]  at scala.reflect.internal.Trees.itraverse(Trees.scala:1341)
[warn]  at scala.reflect.internal.Trees.itraverse$(Trees.scala:1211)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:398)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:149)
[warn]  at scala.reflect.internal.Trees.$anonfun$itraverse$1(Trees.scala:1219)
[warn]  at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
[warn]  at scala.reflect.internal.Trees.traverseMemberDef$1(Trees.scala:1214)
[warn]  at scala.reflect.internal.Trees.itraverse(Trees.scala:1338)
[warn]  at scala.reflect.internal.Trees.itraverse$(Trees.scala:1211)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:398)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:149)
[warn]  at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$2(Trees.scala:2498)
[warn]  at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
[warn]  at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$1(Trees.scala:2498)
[warn]  at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
[warn]  at scala.reflect.internal.Trees.traverseComponents$1(Trees.scala:1243)
[warn]  at scala.reflect.internal.Trees.itraverse(Trees.scala:1341)
[warn]  at scala.reflect.internal.Trees.itraverse$(Trees.scala:1211)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:398)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:149)
[warn]  at scala.reflect.internal.Trees.$anonfun$itraverse$1(Trees.scala:1219)
[warn]  at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
[warn]  at scala.reflect.internal.Trees.traverseMemberDef$1(Trees.scala:1214)
[warn]  at scala.reflect.internal.Trees.itraverse(Trees.scala:1338)
[warn]  at scala.reflect.internal.Trees.itraverse$(Trees.scala:1211)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:398)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:149)
[warn]  at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$2(Trees.scala:2498)
[warn]  at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
[warn]  at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$1(Trees.scala:2498)
[warn]  at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
[warn]  at scala.reflect.internal.Trees.itraverse(Trees.scala:1337)
[warn]  at scala.reflect.internal.Trees.itraverse$(Trees.scala:1211)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
[warn]  at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument$traverser$4$.traverse(DocumentOps.scala:398)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument.$anonfun$toDocument$1(DocumentOps.scala:401)
[warn]  at scala.meta.internal.ReflectionToolkit$CompilationUnitCache.$anonfun$getOrElse$5(ReflectionToolkit.scala:91)
[warn]  at scala.Option.getOrElse(Option.scala:121)
[warn]  at scala.meta.internal.ReflectionToolkit$CompilationUnitCache.getOrElse(ReflectionToolkit.scala:90)
[warn]  at scala.meta.internal.semanticdb.DocumentOps$XtensionCompilationUnitDocument.toDocument(DocumentOps.scala:14)
[warn]  at scala.meta.internal.SemanticdbPipeline$ComputeSemanticdbComponent$ComputeSemanticdbPhase.apply(ScalahostPipeline.scala:55)
[warn]  at scala.tools.nsc.Global$GlobalPhase.$anonfun$applyPhase$1(Global.scala:426)
[warn]  at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:419)
[warn]  at scala.tools.nsc.Global$GlobalPhase.$anonfun$run$1(Global.scala:390)
[warn]  at scala.tools.nsc.Global$GlobalPhase.$anonfun$run$1$adapted(Global.scala:390)
[warn]  at scala.collection.Iterator.foreach(Iterator.scala:929)
[warn]  at scala.collection.Iterator.foreach$(Iterator.scala:929)
[warn]  at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
[warn]  at scala.tools.nsc.Global$GlobalPhase.run(Global.scala:390)
[warn]  at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1431)
[warn]  at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1416)
[warn]  at scala.tools.nsc.Global$Run.compileSources(Global.scala:1412)
[warn]  at scala.tools.nsc.Global$Run.compile(Global.scala:1515)
[warn]  at xsbt.CachedCompiler0.run(CompilerInterface.scala:116)
[warn]  at xsbt.CachedCompiler0.run(CompilerInterface.scala:95)
[warn]  at xsbt.CompilerInterface.run(CompilerInterface.scala:26)
[warn]  at sun.reflect.GeneratedMethodAccessor130.invoke(Unknown Source)
[warn]  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[warn]  at java.lang.reflect.Method.invoke(Method.java:498)
[warn]  at sbt.compiler.AnalyzingCompiler.call(AnalyzingCompiler.scala:107)
[warn]  at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:53)
[warn]  at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:47)
[warn]  at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1.apply$mcV$sp(MixedAnalyzingCompiler.scala:50)
[warn]  at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1.apply(MixedAnalyzingCompiler.scala:50)
[warn]  at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1.apply(MixedAnalyzingCompiler.scala:50)
[warn]  at sbt.compiler.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:74)
[warn]  at sbt.compiler.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:49)
[warn]  at sbt.compiler.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:64)
[warn]  at sbt.compiler.IC$$anonfun$compileInternal$1.apply(IncrementalCompiler.scala:160)
[warn]  at sbt.compiler.IC$$anonfun$compileInternal$1.apply(IncrementalCompiler.scala:160)
[warn]  at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:66)
[warn]  at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:64)
[warn]  at sbt.inc.IncrementalCommon.cycle(IncrementalCommon.scala:32)
[warn]  at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:72)
[warn]  at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:71)
[warn]  at sbt.inc.Incremental$.manageClassfiles(Incremental.scala:99)
[warn]  at sbt.inc.Incremental$.compile(Incremental.scala:71)
[warn]  at sbt.inc.IncrementalCompile$.apply(Compile.scala:54)
[warn]  at sbt.compiler.IC$.compileInternal(IncrementalCompiler.scala:160)
[warn]  at sbt.compiler.IC$.incrementalCompile(IncrementalCompiler.scala:138)
[warn]  at sbt.Compiler$.compile(Compiler.scala:155)
[warn]  at sbt.Compiler$.compile(Compiler.scala:141)
[warn]  at sbt.Defaults$.sbt$Defaults$$compileIncrementalTaskImpl(Defaults.scala:913)
[warn]  at sbt.Defaults$$anonfun$compileIncrementalTask$1.apply(Defaults.scala:904)
[warn]  at sbt.Defaults$$anonfun$compileIncrementalTask$1.apply(Defaults.scala:902)
[warn]  at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
[warn]  at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
[warn]  at sbt.std.Transform$$anon$4.work(System.scala:63)
[warn]  at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
[warn]  at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
[warn]  at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[warn]  at sbt.Execute.work(Execute.scala:237)
[warn]  at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[warn]  at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[warn]  at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[warn]  at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[warn]  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[warn]  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[warn]  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[warn]  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[warn]  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[warn]  at java.lang.Thread.run(Thread.java:745)
[warn] one warning found
```

**Comment by MasseGuillaume** [comment]
```scala
@js.native
@JSImport("codemirror", JSImport.Namespace)
object CodeMirror extends js.Object {
  def on[T <: Event](obj: js.Object, t: String, f: js.Function1[T, _]): Unit = js.native
}
```

```
[warn] failed to generate semanticdb for /home/gui/scastie/client/src/main/scala/codemirror/CodeMirror.scala:
[warn] org.scalameta.UnreachableError: this code path should've been unreachable
[warn] where tpe = scala.scalajs.js.Function1[T, _]
[warn]  at org.scalameta.UnreachableError$.raise(package.scala:40)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.printType(PrinterOps.scala:406)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$printType$2(PrinterOps.scala:373)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$printType$2$adapted(PrinterOps.scala:361)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$printType$1(PrinterOps.scala:344)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$printType$1$adapted(PrinterOps.scala:340)
[warn]  at scala.collection.immutable.List.foreach(List.scala:389)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.wrapped$1(PrinterOps.scala:340)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.printType(PrinterOps.scala:361)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.printType(PrinterOps.scala:385)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.processTreePrinting(PrinterOps.scala:801)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.printTree(PrinterOps.scala:415)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.$anonfun$print$1(Printers.scala:33)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.$anonfun$print$1$adapted(Printers.scala:17)
[warn]  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
[warn]  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
[warn]  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
[warn]  at scala.tools.nsc.ast.Printers$TreePrinter.print(Printers.scala:17)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.super$print(PrinterOps.scala:55)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$print$1(PrinterOps.scala:60)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.$anonfun$print$1$adapted(PrinterOps.scala:52)
[warn]  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
[warn]  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
[warn]  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
[warn]  at scala.meta.internal.semanticdb.PrinterOps$SyntheticCodePrinter.print(PrinterOps.scala:52)
[warn]  at scala.meta.internal.semanticdb.PrinterOps.showSynthetic(PrinterOps.scala:22)
[warn]  at scala.meta.internal.semanticdb.PrinterOps.showSynthetic$(PrinterOps.scala:19)
[warn]  at scala.meta.internal.SemanticdbPlugin.showSynthetic(ScalahostPlugin.scala:9)
[warn]  at scala.meta.internal.semanticdb.PrinterOps.showSynthetic(PrinterOps.scala:17)
...
```

**Comment by MasseGuillaume** [comment]
```scala
import scala.io.AnsiColor

object AnsiColorFormatter extends AnsiColor {

  private val colors = Map(
    BLACK -> "ansi-color-black"
  )
}
```

```
[warn] org.scalameta.UnreachableError: this code path should've been unreachable
[warn] where tpe = String("\u001B[30m")
```

**Comment by MasseGuillaume** [comment]
this one from a component in scala-js-react:

```
 where tpe = japgolly.scalajs.react.raw.ReactComponent[_ <: scala.scalajs.js.Object, _ <: scala.scalajs.js.Object] with japgolly.scalajs.react.component.Scala.Vars[com.olegych.scastie.client.components.Scastie,com.olegych.scastie.client.ScastieState,com.olegych.scastie.client.ScastieBackend]
```

**Comment by olafurpg** [comment]
This has been fixed in scalameta 2.0.0-RC3 and scalafix 0.5.0-RC2
